CodeEdit/Features/Settings/Pages/LocationsSettings/Models/LocationsSettings.swift

```diff
@@ -0,0 +1,52 @@
+//
+//  LocationsSettings.swift
+//  CodeEdit
+//
+//  Created by Raymond Vleeshouwer on 24/06/23.
+//
+
+import Foundation
+
+extension SettingsData {
+
+    struct LocationsSettings: Codable, Hashable {
+        /// The URL of the `themes` folder
+        var themesURL: URL = FileManager.default
+            .homeDirectoryForCurrentUser
+            .appendingPathComponent("Library/Application Support/CodeEdit")
+            .appendingPathComponent("themes", isDirectory: true)
+
+        /// The URL of the `Extensions` folder
+        var extensionsURL: URL = FileManager.default
+            .homeDirectoryForCurrentUser
+            .appendingPathComponent("Library/Application Support/CodeEdit")
+            .appendingPathComponent("Extensions", isDirectory: true)
+
+        /// The URL of the `settings.json` file
+        var settingsURL: URL = FileManager.default
+            .homeDirectoryForCurrentUser
+            .appendingPathComponent("Library/Application Support/CodeEdit")
+            .appendingPathComponent("settings.json", isDirectory: true)
+
+        /// Default Initializer
+        init() {}
+
+        /// Explicit decoder init for setting default values when key is not present in `JSON`
+        init(from decoder: Decoder) throws {
```

Comment:
Storing this in the `settings.json` file makes it seem like we're supporting customizing the settings/themes/extensions folders when that's not the case. It's outside the scope of this PR. This struct should be removed or made into an `enum` with static properties for the URLs defined here.